kmapper/kmapper.py

```diff
@@ -827,6 +827,63 @@ def data_from_cluster_id(self, cluster_id, graph, data):
         else:
             return np.array([])
 
+    def clusters_from_cover(self, cube_ids, graph):
+        """Returns the clusters and their members from the subset of the cover spanned by the given cube_ids
+  
+          Parameters
+          ----------
+          cube_ids : list of int
+              List of hypercube indices.
+          graph : dict
+              The resulting dictionary after applying map().
+  
+          Returns
+          -------
+          nodes : dict
+              cluster membership indexed by cluster ID (subset of `graph["nodes"]`).
+  
+        """
+        clusters = {}
+        cluster_id_prefixes = tuple(["cube"+str(i)+"_" for i in cube_ids])
+        for cluster_id, cluster_members in graph["nodes"].items():
+            if cluster_id.startswith(cluster_id_prefixes):
+                clusters[cluster_id] = cluster_members
+        return clusters
+
+    def nearest_clusters(self, newdata, nodes, data, nn):
+        """Returns the clusters nearest to the `newdata` using the given NearestNeighbors algorithm
+  
+          Parameters
+          ----------
+          newdata : Numpy array
+              New dataset. Accepts both 1-D and 2-D array.
+          nodes : dict
```

Comment:
In theory, this could be _all_ of the nodes from a graph, right? It would be horribly inefficient because there's no way that a point would be close to a cluster that wasn't even in its open set, but still.

If I'm thinking correctly, if it's just for efficiency, then `nearest_clusters` could itself receive the full graph, and itself call `clusters_from_cover` before looping over the nodes to get cluster_members.

Am I thinking correctly?

Also, I lean towards standardizing on replacing `clusters` with `nodes` throughout.

And eventually, but not now, replacing `cube` with `openset` throughout;

if so, that would look something like:

```python
def nearest_nodes(self, newdata, graph, data, nn):
   cube_ids = cover.find(newdata)
   nodes = self.find_nodes(graph, cube_ids)
   
   # then the rest unchanged...
   for cluster_id, cluster_members in nodes.items():
        cluster_data = data[cluster_members]
        nn_data.append(cluster_data)
        nn_cluster_ids.append([cluster_id]*len(cluster_data))
   nn_data = np.vstack(nn_data)
   nn_cluster_ids = np.concatenate(nn_cluster_ids)
   nn.fit(nn_data)
   nn_ids = nn.kneighbors(newdata, return_distance=False)
   return np.unique(nn_cluster_ids[nn_ids])
```